### PR TITLE
Add circle grid to objdetect benchmark

### PR DESCRIPTION
**Pull request by AleksandrPanov**
--configuration=generate_run --board_x=3 --path=res_acircle --synthetic_object=circle_asym_grid
test
image

--configuration=generate_run --board_x=6 --path=res_circle --synthetic_object=circle_sym_grid
test
image

### CODE REVIEW

**Comment by asmorkalov** [comment]
Current 4.x result:
```
python3 ./objdetect_benchmark.py --configuration=generate_run --board_x=3 --path=res_acircle --synthetic_object=circle_asym_grid
```
```
                            category  detected circle_board  total detected circle_board  total circle_board  average detected error circle_board
                     _none_none_blur               1.000000                          216                 216                             0.007245
               _none_none_gaussNoise               1.000000                          216                 216                             0.014082
                     _none_none_none               1.000000                          216                 216                             0.006783
              _none_undistorted_blur               1.000000                          216                 216                             0.190082
        _none_undistorted_gaussNoise               1.000000                          216                 216                             0.189071
              _none_undistorted_none               1.000000                          216                 216                             0.188314
              _perspective_none_blur               0.029321                           19                 648                             0.125292
        _perspective_none_gaussNoise               0.029321                           19                 648                             0.118601
              _perspective_none_none               0.029321                           19                 648                             0.120631
       _perspective_undistorted_blur               0.006173                            4                 648                             0.351288
 _perspective_undistorted_gaussNoise               0.006173                            4                 648                             0.356720
       _perspective_undistorted_none               0.006173                            4                 648                             0.359177
                                 all               0.263310                         1365                5184                             0.102446
total_time:  5.190489999999999
```

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 1]
```diff
@@ -635,20 +647,132 @@ def __check_chessboard(self, synthetic_chessboard, corners):
...
+        if transform_object.name != "":
+            self.history.append(transform_object.name)
+        return self
+
+    def show(self, wait_key=0):
+        assert self.image is not None
+        image = np.copy(self.image)
+        circle_centers = self.circle_centers.reshape(-1, 1, 2)
+        cv.aruco.drawDetectedCornersCharuco(image, circle_centers)
+        cv.imshow("SyntheticCircleGrid", image)
```
Comment (asmorkalov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py]: drawDetectedCornersCharuco is not relevant here.

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 2]
```diff
@@ -635,20 +647,132 @@ def __check_chessboard(self, synthetic_chessboard, corners):
...
+        parameters = cv.SimpleBlobDetector_Params()
+        # parameters.filterByArea = False
+        detector = cv.SimpleBlobDetector_create(parameters)
+        global total_time
+        start = datetime.datetime.now()
+        ret, circle_centers = cv.findCirclesGrid(gray, synthetic_circle_board.board_size, flags=grid_type,
+                                                 blobDetector=detector)
+        total_time += (datetime.datetime.now() - start).total_seconds()
+        # if ret is False:
+        #     circle_centers = None
```
Comment (asmorkalov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py]: Please return it back.

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py (hunk 1 above)
Comment (AleksandrPanov) [reply python_benchmarks/objdetect_benchmark/objdetect_benchmark.py]: fixed

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py (hunk 2 above)
Comment (AleksandrPanov) [reply python_benchmarks/objdetect_benchmark/objdetect_benchmark.py]: fixed and updated benchmark result

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**